data/scripts/git-init-submodules.py

```diff
@@ -0,0 +1,125 @@
+
+# This file is part of the Soletta Project
+#
+# Copyright (C) 2015 Intel Corporation. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#   * Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#   * Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in
+#     the documentation and/or other materials provided with the
+#     distribution.
+#   * Neither the name of Intel Corporation nor the names of its
+#     contributors may be used to endorse or promote products derived
+#     from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import configparser
+import os
+import subprocess
+import sys
+
+def run_command(cmd):
+    try:
+        output = subprocess.check_output(cmd, stderr=subprocess.STDOUT,
+                                         shell=True, universal_newlines=True)
+        return output.replace("\n", "").strip(), True
+    except subprocess.CalledProcessError as e:
+        return e.output, False
+
+def submodule_clone(name, url, branch, depth, dest, sparserules):
+    if os.path.exists(dest):
+        return None, True
+
+    print("GIT: clonning submodule %s ..." % name, end="")
+    sys.stdout.flush()
+    out, status = run_command("git clone {url} {branch} {depth} --bare {dest}".format(
+        url=url, branch=branch, depth=depth, dest=dest))
+
+    if sparserules:
+        submodule_config = "%s/config" % dest
+        rules_file = "%s/info/sparse-checkout" % dest
+
+        config = configparser.ConfigParser()
+        config.read(submodule_config)
+        config.set("core", "sparsecheckout", 'true')
+        config.set("core", "bare", 'false')
+        with open(submodule_config, "w") as fd:
+            config.write(fd)
+
+        content = ""
+        for ln in sparserules.split():
+            content += "%s\n" % ln
+
+        with open(rules_file, "w") as fd:
+            fd.write(content)
+
+    run_command("git submodule init .")
```

Comment:
Does that `.` mean that the script must be run from a particular path? If so, please assert that before doing anything else.